background-on-text-command-line-terminal-applications-and-interfaces.md

```diff
@@ -4,30 +4,30 @@ Background on Text / Command-line / Terminal Applications and Interfaces
 
 Since many assistive technology analysis techniques depend upon discerning the location of the text input cursor, terminal application use of “soft cursors” and “highlight bars” may bypass those analysis techniques and cause failures of success criteria.
 
-<div class="note">It is outside of the scope of this document to define WCAG techniques for non-web ICT. These examples are simply raised here to illustrate how WCAG 2.2 success criteria can be applied to this older class of applications that pre-date the Web.</div>
+<div class="note">It is outside of the scope of this document to define WCAG techniques for non-web ICT. These examples are simply illustrations of how WCAG 2.2 success criteria can be applied to this class of non-web software applications.</div>
 
-The way to think about “[accessibility supported](http://w3c.github.io/wcag2ict/#wcag2ict-def_accessibility-supported)” and “[programmatically determined](http://w3c.github.io/wcag2ict/#wcag2ict-def_programmaticallydetermined)” may seem a little different for text applications, but the definitions are unchanged. Because assistive technologies developed analysis techniques to recognize many forms of tabular layout, column headers, and section headings, such recognized layouts are by definition “accessibility supported”. Where assistive technology is able to extract and present that information, it is “programmatically determinable”—even though no explicit markup or API was used to make it so. While such assistive technology analysis techniques are not used for supporting newer technologies, the fact that such previous analysis work was done allows us to apply WCAG 2.2 to text / command-line applications.
+The way to think about “accessibility supported” and “programmatically determined” may seem a little different for text applications, but the definitions are unchanged. Because assistive technologies developed analysis techniques to recognize many forms of tabular layout, column headers, and section headings, such recognized layouts are by definition “accessibility supported”. Where assistive technology is able to extract and present that information, it is “programmatically determinable”—even though no explicit markup or API was used to make it so. While such assistive technology analysis techniques are not used for supporting newer technologies, the fact that such previous analysis work was done allows us to apply WCAG 2.2 to text / command-line applications.
```

Comment:
Because of @mraccess77's [comment on an earlier pull request](https://github.com/w3c/wcag2ict/pull/154#issuecomment-1536358120), I believe this paragraph is inaccurate from the second sentence onward:

> The way to think about “accessibility supported” and “programmatically determined” may seem a little different for text applications, but the definitions are unchanged. Because assistive technologies developed analysis techniques to recognize many forms of tabular layout, column headers, and section headings, such recognized layouts are by definition “accessibility supported”. Where assistive technology is able to extract and present that information, it is “programmatically determinable”—even though no explicit markup or API was used to make it so. While such assistive technology analysis techniques are not used for supporting newer technologies, the fact that such previous analysis work was done allows us to apply WCAG 2.2 to text / command-line applications.

We should replace it with something like this:

> The way to think about “accessibility supported” and “programmatically determined” may seem a little different for text applications, but the definitions are unchanged. <ins>Unlike the semantic objects of graphical user interfaces and web pages, the output of text-based applications consists of plain text. A terminal emulator acts as the user agent for text-based applications; it might render some content such as escape codes as semantic elements, but otherwise exposes only lines of text to assistive technology. Where assistive technology is able to interpret the text and any semantic objects accurately,</ins> the content is “programmatically determinable”—even though no explicit markup or API was used to make it so. <del>While such assistive technology analysis techniques are not used for supporting newer technologies, the fact that such previous analysis work was done allows us to apply WCAG 2.2 to text / command-line applications.</del>

As a sighted accessibility professional, my screen reader experience is adequate for most web and native app evaluations, but I'm rudimentary when it comes to terminal applications specifically. So if this draft is not quite right, please improve it.

@maryjom, regarding [your comment in the earlier pull request](https://github.com/w3c/wcag2ict/pull/154#issuecomment-1536693667):

> I don't think WCAG2ICT can make comment on testing techniques and guidance, as our scope doesn't permit us to come up with techniques.

True, but I wasn't actually talking about the analysis that a tester would perform. I was quoting the existing text, where "assistive technologies developed analysis techniques to recognize many forms of tabular layout, column headers, and section headings." I understand this to refer to (hypothetical) screen reader heuristic repair techniques for text applications, akin to the heuristics that screen readers use to repair incomplete and malformed web pages.

Reply:
+1 to Mitch's proposed changes.

Reply:
Alas the comment-on-a-specific-line feature in GitHub is not very accessible (general discussion thread comments are OK, and also supported in the `gh` CLI, as it happens); we discussed your suggestions on our call and incorporated them almost exactly, in 534e2165790c2619c36b91d90e5949f4c10e2917; thanks!